File: src/net/intensicode/tools/BitmapFontMaker.java
package net.intensicode.tools;

import net.intensicode.tools.bfm.*;

import java.io.IOException;

public final class BitmapFontMaker
    {
    public static void main( final String[] aArgs ) throws IOException
        {
        final UiController controller = new UiController();
        final FontRenderer fontRenderer = new FontRenderer();
        final UiBuilder uiBuilder = new UiBuilder( controller, fontRenderer );
        final MainFrame frame = new MainFrame( uiBuilder );
        frame.centerAndResizeTo( 80, 80 );
        frame.setVisible( true );
        }
    }

File: src/jruby/bitmap_font_maker.rb
$LOAD_PATH << File.dirname(__FILE__)

require 'java'

require 'bfm/font_renderer'
require 'bfm/main_frame'
require 'bfm/ui_controller'
require 'bfm/ui_builder'

renderer = BFM::FontRenderer.new
controller = BFM::UiController.new(renderer)
builder = BFM::UiBuilder.new(controller, renderer)
frame = BFM::MainFrame.new(builder)
frame.center_and_resize_to(80, 80)
frame.set_visible(true)
